src/policies/previous_guidance.rs

```diff
@@ -0,0 +1,1033 @@
+
+                self.add_plan(proposed_masking_end_time, move |context| {
+                    context.set_person_property(person_id, MaskingStatus, false);
+                    trace!("Person {person_id} is now no longer masking");
+                });
+            }
+        }
+    }
+
+    fn schedule_test(
+        &mut self,
+        person_id: PersonId,
+        intervention_policy_parameters: InterventionPolicyParameters,
+    ) {
+        self.add_plan(
+            self.get_current_time() + intervention_policy_parameters.delay_to_retest,
+            move |context| {
+                if context.get_person_property(person_id, PresentingWithSymptoms) {
+                    context.administer_test_and_schedule_followup(
+                        person_id,
+                        intervention_policy_parameters,
+                    );
+                }
+            },
+        );
+    }
+
+    fn administer_test_and_schedule_followup(
+        &mut self,
+        person_id: PersonId,
+        intervention_policy_parameters: InterventionPolicyParameters,
+    ) {
+        let last_test_result = self.get_person_property(person_id, LastTestResult);
+        let mut retest = false;
+
+        if self.get_person_property(person_id, InfectionStatus) == InfectionStatusValue::Infectious
+        {
+            if self.sample_bool(
+                PreviousPolicyRng,
+                intervention_policy_parameters.test_sensitivity,
+            ) {
+                // A known positive doesn't require a retest, keeping default `false`
+                self.set_person_property(person_id, LastTestResult, Some(true));
+            } else {
+                if last_test_result.is_none() {
+                    retest = true;
+                }
+                self.set_person_property(person_id, LastTestResult, Some(false));
+            }
+        } else {
+            if last_test_result.is_none() {
+                retest = true;
+            }
+            self.set_person_property(person_id, LastTestResult, Some(false));
+        }
+        trace!("Person {person_id} was tested");
+
+        if retest {
+            self.schedule_test(person_id, intervention_policy_parameters);
+        }
+    }
+
+    fn handle_symptom_resolution(
+        &mut self,
+        person_id: PersonId,
+        intervention_policy_parameters: InterventionPolicyParameters,
+    ) {
+        if self
+            .get_person_property(person_id, LastTestResult)
+            .unwrap_or(false)
+        {
+            if let Some(symptom_record) = self.get_person_property(person_id, SymptomRecord) {
+                let isolation_start_time = symptom_record.symptom_start
+                    + intervention_policy_parameters.isolation_delay_period;
+                let minimum_isolation_time = if symptom_record.severe {
+                    intervention_policy_parameters.moderate_symptom_isolation_duration
+                        + isolation_start_time
```

Comment:
```suggestion
                        + isolation_start_time - intervention_policy_parameters.isolation_delay_period
```
This is probably not the most natural / best way to do this, but I am making this suggestion because I want to flag that `moderate_symptom_isolation_duration` and `mild_symptom_isolation_duration` are defined in terms of time since symptom onset. (I.e., if a person delays entering isolation, there's no biological or public health reason for them to have to stay in isolation a longer amount equivalent to the delay).